### PR TITLE
PLAT-576 Add Notification when delegating Workflow Tasks

### PR DESCRIPTION
**Pull request by daniel-klose**

### CODE REVIEW

**Comment by daniel-klose** [comment]
image

**Comment by daniel-klose** [comment]
image

**Comment by daniel-klose** [comment]
image

**Comment by daniel-klose** [comment]
image

File: platform-workflow-module/src/main/java/com/softicar/platform/workflow/module/workflow/task/WorkflowTaskNotificationSubmitter.java [hunk 1]
```diff
@@ -4,23 +4,32 @@
...
 	private final AGWorkflowTask task;
+	private final AGUser user;
 
 	public WorkflowTaskNotificationSubmitter(AGWorkflowTask task) {
 
 		this.task = task;
+		this.user = task.getUser();
+	}
+
+	public WorkflowTaskNotificationSubmitter(AGWorkflowTask task, AGUser user) {
```
Comment (alexander-schmidt) [inline platform-workflow-module/src/main/java/com/softicar/platform/workflow/module/workflow/task/WorkflowTaskNotificationSubmitter.java:21]: I think we should take a slightly different approach here, and replace this new constructor with a method:
```
public WorkflowTaskNotificationSubmitter setNotificationRecipient(AGUser notificationRecipient) {
    this.notificationRecipient = Objects.requireNonNull(notificationRecipient);
    return this;
}
```
The above constructor would still assign `this.notificationRecipient = task.getUser();`, ofc.

File: platform-workflow-module/src/main/java/com/softicar/platform/workflow/module/workflow/task/delegation/WorkflowTaskDelegationSaveHook.java [hunk 2]
```diff
@@ -0,0 +1,19 @@
+package com.softicar.platform.workflow.module.workflow.task.delegation;
+
+import com.softicar.platform.emf.table.listener.IEmfSaveHook;
+import com.softicar.platform.workflow.module.workflow.task.WorkflowTaskNotificationSubmitter;
+import java.util.Collection;
+
+public class WorkflowTaskDelegationSaveHook implements IEmfSaveHook<AGWorkflowTaskDelegation> {
```
Comment (alexander-schmidt) [inline platform-workflow-module/src/main/java/com/softicar/platform/workflow/module/workflow/task/delegation/WorkflowTaskDelegationSaveHook.java:7]: I think this needs unit test coverage.

**Review by alexander-schmidt — CHANGES_REQUESTED** [review]

File: platform-workflow-module/src/main/java/com/softicar/platform/workflow/module/workflow/task/delegation/WorkflowTaskDelegationSaveHook.java (hunk 2 above)
Comment (daniel-klose) [reply platform-workflow-module/src/main/java/com/softicar/platform/workflow/module/workflow/task/delegation/WorkflowTaskDelegationSaveHook.java:7]: I dont think so.
The saveHooks themselves should already have unit tests, and all im doing here is using an already tested class.
As such, I dont really know what I should test here specifically

Comment (daniel-klose) [reply platform-workflow-module/src/main/java/com/softicar/platform/workflow/module/workflow/task/delegation/WorkflowTaskDelegationSaveHook.java:7]: But I added another test case in WorkflowTaskNotificationSubmitterTest

Comment (alexander-schmidt) [reply platform-workflow-module/src/main/java/com/softicar/platform/workflow/module/workflow/task/delegation/WorkflowTaskDelegationSaveHook.java:7]: I think that's exactly what should be tested.

**Review by alexander-schmidt — APPROVED** [review]

**Pull request merged**